### ISSUE TITLE
User friendly error handling of wrong Github access token

### ISSUE THREAD

**Issue by griable** [foo-software/lighthouse-check-action#23]
Hi there,

I'm in love with your tool but I'd like to make a suggestion.

When dealing with a wrong Github access token, this is the error I'm getting.

```
Run foo-software/lighthouse-check-action@master
  with:
    accessToken: ***
    author: username
    branch: refs/pull/26/merge
    outputDirectory: /tmp/artifacts
    urls: <url>
[@octokit/rest] `const Octokit = require("@octokit/rest")` is deprecated. Use `const { Octokit } = require("@octokit/rest")` instead
lighthouse-check: Auditing (1/1) https://www.google.com/
lighthouse-check: LighthouseCheckError: Not Found
    at _default (/home/runner/work/_actions/foo-software/lighthouse-check-action/master/node_modules/@foo-software/lighthouse-check/dist/postPrComment.js:100:13)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async /home/runner/work/_actions/foo-software/lighthouse-check-action/master/node_modules/@foo-software/lighthouse-check/dist/lighthouseCheck.js:198:11 {
  code: 'ERROR_UNEXPECTED_RESPONSE',
  data: {
    message: 'Not Found',
    documentation_url: 'https://developer.github.com/v3'
  }
}
lighthouse-check:
 LighthouseCheckError: Not Found
    at _default (/home/runner/work/_actions/foo-software/lighthouse-check-action/master/node_modules/@foo-software/lighthouse-check/dist/postPrComment.js:100:13)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async /home/runner/work/_actions/foo-software/lighthouse-check-action/master/node_modules/@foo-software/lighthouse-check/dist/lighthouseCheck.js:198:11
```

IMHO, it's a bit cryptic and doesn't really help in identifying the root cause of the issue.

So a nice optimisation might be to make this more explicit, what do you think?

PS: Thanks again for your products!

**Comment by adamhenson** [comment]
Thanks @griable, I'll take this on when I have a moment.

**Comment by lukasoppermann** [comment]
Running into this issue as well. Can you give me a hint what the issue is? I guess I need a specific permission set on the token?

**Comment by adamhenson** [comment]
Hello @lukasoppermann this issue relates to [`accessToken`](https://github.com/foo-software/lighthouse-check-action#inputs) and is labelled an "enhancement" because the ask is for better error messaging. Currently we log the raw error from the [GitHub API](https://developer.github.com/v3/#authentication) which is occurring because we send a request to the GitHub API in order to post comments on PRs. It's unfortunate that GitHub API responds with a 404 (not found) rather than a more appropriate 403 (forbidden)... which would make the messaging more clear. Eventually we can perhaps do more from our end by at least log what part of the process is failing (PR comments).

**tl;dr** - to fix this error you should take a deeper look on what value you're populating for `accessToken`. As documented, you can find [details about how to generate this token here](https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token).

Does that make sense @lukasoppermann? Let me know if you have any further issues.

**Comment by adamhenson** [comment]
Yes, the user who owns the access token would need to have write access to the repo.

**Comment by sdwvit** [comment]
still an issue

**Comment by adamhenson** [comment]
@sdwvit - please describe your issue in detail. This is an implementation problem, not an issue with the module. The error is correctly logged.

If you can provide details of your problem, perhaps I can help.

**Comment by sdwvit** [comment]
@adamhenson 
I am using a separate github user to post messages from integration, and I had this new repo I integrated lighthouse check action into. Couldn't figure out why my CI would spit:
```
lighthouse-check: LighthouseCheckError: Not Found
    at _default (/home/runner/work/_actions/foo-software/lighthouse-check-action/master/node_modules/@foo-software/lighthouse-check/dist/postPrComment.js:119:13)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async /home/runner/work/_actions/foo-software/lighthouse-check-action/master/node_modules/@foo-software/lighthouse-check/dist/lighthouseCheck.js:201:11 {
  code: 'ERROR_UNEXPECTED_RESPONSE',
  data: {
    message: 'Not Found',
    documentation_url: 'https://docs.github.com/rest'
  }
}
```
I had token in place, it's working on other repo, so I was super confused. 
Then I found this issue and realized I forgot to add user as collaborator.

**Comment by adamhenson** [comment]
This module is logging the response directly from the GitHub API. If you have a more ideal proposal, please detail here or offer a pull request. This is an open-source project.
